tango/workspace.py

```diff
@@ -228,6 +85,23 @@ class Workspace(Registrable):
     def __init__(self):
         self._delayed_cleanup_temp_dirs: List[TemporaryDirectory] = []
 
+    def __getstate__(self):
+        """
+        We override `__getstate__()` to customize how instances of this class are pickled
+        since we don't want to persist certain attributes.
+        """
+        out = {k: v for k, v in self.__dict__.items() if k not in {"_delayed_cleanup_temp_dirs"}}
+        out["_delayed_cleanup_temp_dirs"] = []
+        return out
+
+    @property
+    def url(self) -> str:
+        """
+        Get a URL for the workspace that can be used to instantiate the same workspace
+        using :meth:`.from_url()`.
+        """
+        raise NotImplementedError
+
     @classmethod
     def from_url(cls, url: str) -> "Workspace":
```

Comment:
The fact that this takes URLs as well as local paths seems worth mentioning in the docs.

Reply:
In addition to this?
https://github.com/allenai/tango/blob/9e6c3ee3caf02f5b3f011014377359473a8f3d10/tango/workspace.py#L108

Reply:
I checked the code to make sure that `"/tmp/workspace"` also still works. It wasn't clear to me from the docs.